lib/ansible/modules/cloud/amazon/ec2_lc.py

```diff
@@ -225,79 +325,133 @@ def create_launch_config(connection, module):
         for volume in volumes:
             if 'device_name' not in volume:
                 module.fail_json(msg='Device name must be set for volume')
-            # Minimum volume size is 1GB. We'll use volume size explicitly set to 0
-            # to be a signal not to create this volume
+            # Minimum volume size is 1GB. We'll use volume size explicitly
+            # set to 0 to be a signal not to create this volume
             if 'volume_size' not in volume or int(volume['volume_size']) > 0:
-                bdm[volume['device_name']] = create_block_device(module, volume)
-
-    lc = LaunchConfiguration(
-        name=name,
-        image_id=image_id,
-        key_name=key_name,
-        security_groups=security_groups,
-        user_data=user_data,
-        block_device_mappings=[bdm],
-        instance_type=instance_type,
-        kernel_id=kernel_id,
-        spot_price=spot_price,
-        instance_monitoring=instance_monitoring,
-        associate_public_ip_address=assign_public_ip,
-        ramdisk_id=ramdisk_id,
-        instance_profile_name=instance_profile_name,
-        ebs_optimized=ebs_optimized,
-        classic_link_vpc_security_groups=classic_link_vpc_security_groups,
-        classic_link_vpc_id=classic_link_vpc_id,
-    )
+                bdm.update(create_block_device_meta(module, volume))
 
-    launch_configs = connection.get_all_launch_configurations(names=[name])
+    launch_configs = connection.describe_launch_configurations(
+        LaunchConfigurationNames=[name]).get('LaunchConfigurations')
     changed = False
-    if not launch_configs:
+    result = {}
+
+    launch_config = {
+        'LaunchConfigurationName': name,
+        'EbsOptimized': ebs_optimized,
+        'PlacementTenancy': placement_tenancy
+    }
+
+    if image_id is not None:
```

Comment:
might be better to have some kind of mapping here

```
convert_list = ['image_id', 'instance_type', 'instance_id', ...]
camel_params = dict((k,v) for k, v in snake_dict_to_camel_dict(module.params).items() if v is not None and k is in convert_list)
```

and then special case any exceptions like IamInstanceProfile and InstanceMonitoring